app/derivative_transformers/kithe/vips_cli_pdf_to_jpeg.rb

```diff
@@ -0,0 +1,120 @@
+require 'tempfile'
+require 'tty/command'
+
+module Kithe
+  # Use the [vips](https://jcupitt.github.io/libvips/) command-line utility (via shell-out)
+  # to transform any image type to a JPG, with a specified maximum width (keeping aspect ratio).
+  #
+  # Requires vips command line utilities `vips` and `vipsthumbnail` and to be installed on your system,
+  # eg `brew install vips`, or apt package `vips-tools`.
+  #
+  # We ALSO apply some additional best practices
+  # for minimizing size when used as an image _in a browser_, such as removing
+  # color profile information. See eg:
+  #  * https://developers.google.com/speed/docs/insights/OptimizeImages
+  #  * http://libvips.blogspot.com/2013/11/tips-and-tricks-for-vipsthumbnail.html
+  #  * https://github.com/jcupitt/libvips/issues/775
+
+  class Kithe::VipsCliPdfToJpeg
+    class_attribute :srgb_profile_path, default: Kithe::Engine.root.join("lib", "vendor", "icc", "sRGB2014.icc").to_s
+    class_attribute :vips_thumbnail_command, default: "vipsthumbnail"
+    class_attribute :vips_command, default: "vips"
+    class_attribute :vips_header_command, default: "vipsheader"
+
+    attr_reader :max_width, :jpeg_q
+
+    def initialize(max_width:nil, jpeg_q: 98)
+      @max_width = max_width
+      @jpeg_q = jpeg_q
+      @border_pixels = 1
+      @cmd = TTY::Command.new(printer: :null)
+
+      if max_width.nil?
+        # https://github.com/libvips/libvips/issues/1179
+        raise ArgumentError.new("thumbnail_mode currently requires a non-nil max_width")
+      end
+    end
+
+    # Will raise TTY::Command::ExitError if the external Vips command returns non-null.
+    def call(original)
+      files = (0..2).map {|x| Tempfile.new(["kithe_vips_pdf", ".jpg"])}
+      # Make a slightly smaller thumbnail to accomodate the border we're adding.
+      width_minus_borders = @max_width - (2 * @border_pixels)
+      # Create and sharpen a thumbnail, then add a black border to it.
+      thumbnail( width_minus_borders, original, files[0])
+      sharpen(   width_minus_borders, files[0], files[1])
+      add_border(width_minus_borders, files[1], files[2])
+      # The width of the thumbnail, including the border,
+      # is now exactly @max_width.
```

Comment:
while Tempfile is theoretically cleaned up by ruby when it gets Garbage Collected, if we know we aren't going to need files[0] or files[1] again, we can make things easier on ruby by calling `files[0].unlink` and `files[1].unlink`

https://ruby-doc.org/stdlib-1.9.3/libdoc/tempfile/rdoc/Tempfile.html#method-i-unlink

Also, nit-picking, but as long as I'm here, it might be less confusing to use individually named local variables. Cleaning up that `map` a bit too...

thumbnail, sharpened_thumb, with_borders_thumb = 3.times.map { Tempfile.new(["kithe_vips_pdf", ".jpg"]) }